lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\RepositoryForms\ConfigResolver;
+
+use Closure;
+use eZ\Publish\API\Repository\Repository;
+use OutOfBoundsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * A repository data loader that uses the sudo() method.
+ *
+ * It comes with parameter handling, either by passing an array of (supported) options
+ * to the constructor, or using the setParam() method.
+ *
+ * Implementations will call load() with a repository callback as an argument.
+ * The repository can be accessed using getRepository().
+ *
+ * ** Use with care**.
+ */
+abstract class ConfigurableSudoRepositoryLoader
+{
+    /**
+     * @var Repository
+     */
+    private $repository;
+
+    /**
+     * @var array
+     */
+    private $params = [];
```

Comment:
```suggestion
    private $params;
```
should it be null or array?